### PR TITLE
[AKS] Add Key Management Service feature for AKS to encrypt the Kubernetes data in etcd using Azure Key Vault with KMS plugin

### PR DESCRIPTION
**Pull request by bingosummer**
**Related command**

az aks create --enable-azure-keyvault-kms --azure-keyvault-kms-key-id {} --azure-keyvault-kms-key-vault-network-access {} --azure-keyvault-kms-key-vault-resource-id {}

**Description**

This PR allows the customer to enable KMS feature for AKS cluster, which [encrypts the Kubernetes data in etcd using Azure Key Vault with Key Management Service (KMS) plugin](https://docs.microsoft.com/en-us/azure/aks/use-kms-etcd-encryption).

**Testing Guide**

**History Notes**

[AKS] `az aks create`: Add Key Management Service feature for AKS to encrypt the Kubernetes data in etcd using Azure Key Vault with KMS plugin

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [x] The PR title and description has followed the guideline in [Submitting Pull Requests](https://github.com/Azure/azure-cli/tree/dev/doc/authoring_command_modules#submitting-pull-requests).

- [x] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

- [x] I adhere to the [Error Handling Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/error_handling_guidelines.md).

### CODE REVIEW

**Comment by yonzhan** [comment]
AKS

File: src/azure-cli/azure/cli/command_modules/acs/managed_cluster_decorator.py [hunk 1]
```diff
@@ -3683,6 +3686,255 @@ def get_defender_config(self) -> Union[ManagedClusterSecurityProfileDefender, No
...
+
+        :return: bool
+        """
+        # read the original value passed by the command
+        enable_azure_keyvault_kms = self.raw_param.get("enable_azure_keyvault_kms")
+        # In create mode, try to read the property value corresponding to the parameter from the `mc` object.
+        if self.decorator_mode == DecoratorMode.CREATE:
+            if (
+                self.mc and
+                self.mc.security_profile and
```
Comment (FumingZhang) [inline src/azure-cli/azure/cli/command_modules/acs/managed_cluster_decorator.py:3703]:
```suggestion
                hasattr(self.mc, "security_profile") and  # backward compatibility
                self.mc.security_profile and
```

to bypass compatibility issue

**Comment by bingosummer** [comment]
https://dev.azure.com/msazure/CloudNativeCompute/_build/results?buildId=58198057&view=results

File: src/azure-cli/azure/cli/command_modules/acs/tests/latest/test_aks_commands.py [hunk 2]
```diff
@@ -6848,3 +6848,573 @@ def test_aks_create_node_resource_group(self, resource_group, resource_group_loc
 
         # delete
         self.cmd('aks delete -g {resource_group} -n {name} --yes --no-wait', checks=[self.is_empty()])
+
+    @live_only()
+    @AllowLargeResponse()
+    @AKSCustomResourceGroupPreparer(random_name_length=17, name_prefix='clitest', location='centraluseuap')
+    def test_aks_create_with_azurekeyvaultkms_public_key_vault(self, resource_group, resource_group_location):
```
[the remark this answers is not in the document]
Comment (FumingZhang) [reply src/azure-cli/azure/cli/command_modules/acs/tests/latest/test_aks_commands.py:6855]: Second [one](https://dev.azure.com/msazure/CloudNativeCompute/_build/results?buildId=58235608&view=results), previous one failed due to feature `Microsoft.ContainerService/AzureKeyVaultKmsPreview` not registered.

[the remark this answers is not in the document]
Comment (bingosummer) [reply src/azure-cli/azure/cli/command_modules/acs/tests/latest/test_aks_commands.py:6855]: I also queued [one](https://dev.azure.com/msazure/CloudNativeCompute/_build/results?buildId=58235663&view=results).

**Review by FumingZhang — APPROVED** [review]
LGTM

**Review by zhoxing-ms — APPROVED** [review]

**Comment by zhoxing-ms** [comment]
@bingosummer Could you please add the description of those new parameters in the PR description. Please note that these contents will be written into the history notes. 
For more details, please refer to [submitting-pull-requests](https://github.com/Azure/azure-cli/tree/dev/doc/authoring_command_modules#submitting-pull-requests)

**Comment by bingosummer** [comment]
@zhoxing-ms 
Updated the PR description. Does that meet the requirements?

**Pull request merged**